### PR TITLE
[acl] Extend the ACL scripts to cover egress ACL

### PR DESCRIPTION
**Pull request by wangxin**
1. Extended the ACL scripts to cover egress ACL testing:
   * Added new code to cover egress ACL configuration and testing
   * Added a acl_facts module for collecting ACL facts including ACL
     counters
   * Added code for checking ACL counters after PTF run
2. Improved the existing ACL scripts:
   * Added more ACL rules and cases for testing the drop action
   * Fixed issue https://github.com/Azure/sonic-mgmt/issues/708
   * Improved the logging and reporting of the PTF script
   * Covered adding ACL table

Signed-off-by: Xin Wang <xinw@mellanox.com>

### Description of PR

Summary:
Fixes # (issue)

This is a big change of the existing ACL scripts:
1. Added case for testing egress ACL
2. Made a lot of improvements to the existing ACL

For more detailed information about this change, please refer to this document for now: https://github.com/wangxin/SONiC/blob/acl/doc/acl/ACL-egress-test-plan.md

### Type of change

- [] Bug fix
- [] Testbed and Framework(new/improvement)
- [*] Test case(new/improvement)

#### How did you do it?
* Made significant changes to the existing ansible and PTF scripts. 
* Added new ACL rules. Optimized the values of the ACL rules.
* Added a new acl_facts ansible module for collecting ACL facts including ACL counters

#### How did you verify/test it?
Tested ingress ACL on mellanox platform using public image.
Tested egress ACL on mellanox platform using internally built image. The public image from master branch will support egress ACL soon. Merging of the PR for adding egress capability is in progress.

#### Supported testbed topology if it's a new test case?
Same as existing ACL scripts, supported topology: t1, t1-lag, t1-64-lag

### Documentation 

For more details, please refer to this document for now: https://github.com/wangxin/SONiC/blob/acl/doc/acl/ACL-egress-test-plan.md

I will update this document to the SONiC wiki.

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@lguohan anyone can review the PR? this is a feature as part of 03/2019 release which was merged and ansible test is required for full functionality guarantee.

File: ansible/roles/test/files/acstests/acltb_test.py [hunk 1]
```diff
@@ -93,225 +103,314 @@ def verify_packet_any_port(self, pkt, ports=[], device_number=0):
...
-        pkt['TCP'].dport = 0x1235
-        exp_pkt['TCP'].dport = 0x1235
-        self.runSendReceiveTest(4, pkt, src_port, exp_pkt, dst_ports, 1)
-
-        # Test #5 - IP protocol match - forwarded
-        pkt = pkt0.copy()
-        exp_pkt = exp_pkt0.copy()
+        direction = ", " + direction
+
+        print("\nPort to sent packets to:")
```
Comment (stcheng) [inline ansible/roles/test/files/acstests/acltb_test.py]: make these two outputs in one line

**Comment by stcheng** [comment]
the change looks good to me. still I wonder if it is possible to split the test into at least three parts:
1. regular ACL test
2. test with interface toggle
3. test with switch reboot

This will make each sub test shorter and easy for re-run. It could be adding some arguments to pick which one or all three to run at the same time.

could you also provide the test result for mellanox platforms for all various topologies?

**Comment by wangxin** [comment]
Thanks Shuotian for the review! Splitting the test into 3 parts is a good idea. I'll update according and push new commits. Test results on Mellanox platforms for various topologies will be attached after I completed the changing.

**Comment by wangxin** [comment]
@stcheng I have updated the acl scripts and splitted into 3 pieces. Could you please help review again?
Attached test results on Mellanox platforms for t1 and t1-lag topologies.

[acl_ansible_log_ingress_egress_t1.log](https://github.com/Azure/sonic-mgmt/files/3061804/acl_ansible_log_ingress_egress_t1.log)
[acl_ansible_log_ingress_egress_t1-lag.log](https://github.com/Azure/sonic-mgmt/files/3061805/acl_ansible_log_ingress_egress_t1-lag.log)

**Pull request merged**